### ISSUE TITLE
BUG: unable to parsing combined file extension

### ISSUE THREAD

**Issue by xshaokun** [neutrinoceros/inifix#148]
Pluto supports two HDF5 data formats `dbl.h5` and `flt.h5`. But the validation doesn't allow the period in parameter names for now. Namely, the following lines will raise `ValueError`,

```
dbl.h5    -1.0  -1
flt.h5    -1.0  -1
```

Adding the period into `_PARAM_NAME_REGEXP` could solve this issue, but I think it might be too lazy for other parameters since the period only appears in the session `Static Grid Output`.

**Comment by neutrinoceros** [comment]
Hi @xshaokun

I've tried reproducing this error with the following script, but it seems to be working correctly
```python
import inifix

data = (
    "[Static Grid Output]\n"
    "dbl.h5    -1.0  -1"
)
conf = inifix.loads(data)
print(conf)
```
output
```
{'Static': {'dbl.h5': [-1, -1]}}
```

Could you show a self-contained reproducer for the error you are getting ?

**Comment by neutrinoceros** [comment]
nevermind, I got it: loading with `inifix.load` instead of `inifix.loads` indeed raises an error at validation

```python
import inifix

data = (
    "[Static]\n"
    "dbl.h5    -1.0  -1"
)
with open("/tmp/data.ini", "w") as fh:
    fh.write(data)

conf = inifix.load("/tmp/data.ini")
print(conf)
```

output
```
ValueError: Invalid schema: received key 'dbl.h5', expected only alphanumeric characters or '-'
```

Well, this is definitely an oversight/blind spot of my validation routine. I agree this should be considered a bug. Let me know if you'd like to issue a PR, otherwise I'm happy to fix this myself !

**Comment by xshaokun** [comment]
Hi @neutrinoceros, this is exactly what I got!

 I think you can fix it sooner and in a better way than mine. Thank you!

**Comment by neutrinoceros** [comment]
@xshaokun I've opened 2 PRs to address this issue
- #151 implements your suggested fix
- #152 adds a `skip_validation` boolean parameter to all IO functions so that if similar issues ever pop up again, we have a builtin workaround so you don't get blocked immediately (but of course reports and PRs are still welcome if this is ever needed !)

I'll give you some time to respond if there's anything missing in this work, and I can cut a release after that.

**Comment by xshaokun** [comment]
@neutrinoceros That's all, and `skip_validation` will be really useful, thanks!

**Comment by neutrinoceros** [comment]
Both patches are now released in version 4.1.0
Thank you again for reporting !
